kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -135,15 +136,17 @@ private PartitionedTableId getRecordTable(SinkRecord record) {
     TableId baseTableId = topicsToBaseTableIds.get(record.topic());
 
     PartitionedTableId.Builder builder = new PartitionedTableId.Builder(baseTableId);
-    if (useMessageTimeDatePartitioning) {
-      if (record.timestampType() == TimestampType.NO_TIMESTAMP_TYPE) {
-        throw new ConnectException(
-            "Message has no timestamp type, cannot use message timestamp to partition.");
-      }
-
-      builder.setDayPartition(record.timestamp());
-    } else {
-      builder.setDayPartitionForNow();
+    if(useIngestionTimePartitioning) {
```

Comment:
This might be wrong. BigQuery support 3 types of partitioning. `useMessageTimeDatePartitioning` is for `Date/timestamp` so we should be able to use `MessageTimeDatePartitioning` when `useIngestionTimePartitioning = False`.
```
You can partition BigQuery tables by:
Ingestion time: Tables are partitioned based on the data's ingestion (load) date or arrival date.
Date/timestamp: Tables are partitioned based on a TIMESTAMP or DATE column.
Integer range: Tables are partitioned based on an integer column.
```